OrbitService/Utils.cpp

```diff
@@ -175,29 +158,117 @@ ErrorMessageOr<std::vector<orbit_grpc_protos::TracepointInfo>> ReadTracepoints()
   return result;
 }
 
-ErrorMessageOr<std::unordered_map<pid_t, double>> GetCpuUtilization() {
-  const std::string cmd = "top -b -n 1 -w512 | sed -n '8, 1000{s/^ *//;s/ *$//;s/  */,/gp;};1000q'";
-  OUTCOME_TRY(top_data, ExecuteCommand(cmd));
-  return GetCpuUtilization(top_data);
+static std::optional<pid_t> ProcEntryToPid(const std::filesystem::directory_entry& entry) {
+  if (!entry.is_directory()) {
+    return std::nullopt;
+  }
+
+  int potential_pid;
+  if (!absl::SimpleAtoi(entry.path().filename().string(), &potential_pid)) {
+    return std::nullopt;
+  }
+
+  if (potential_pid <= 0) {
+    return std::nullopt;
+  }
+
+  return static_cast<pid_t>(potential_pid);
+}
+
+namespace {
+struct EmplaceBackIfHasValue {
+  template <typename T>
+  std::vector<T> operator()(std::vector<T> vec, const std::optional<T>& val) const {
+    if (val.has_value()) {
+      vec.emplace_back(std::move(val.value()));
+    }
+
+    return std::move(vec);
+  }
+};
+}  // namespace
```

Comment:
Is it not possible to use a lambda instead?

Reply:
Yeah, good point. This was used in two different functions before, but I could simplifiy the code and forgot to remove that helper function again. Thanks!